src/app_service/service/chat/service.nim

```diff
@@ -233,9 +233,13 @@ QtObject:
     for chat in chats:
       if (chat.active):
         self.events.emit(SIGNAL_CHAT_CREATED, CreatedChatArgs(chat: chat))
-
-    for i, msg in messages:
+      
+    var i = 0
+    for msg in messages:
       self.events.emit(SIGNAL_SENDING_SUCCESS, MessageSendingSuccess(message: msg, chat: chats[i]))
+      # messages can be received from one chat (e.g. sending several images)
+      if chats.len - 1 > i:
+        i.inc()
```

Comment:
I feel like this approach might cause problems as well, since just incrementing on each message is a bit arbitrary.
Shouldn't we actually do a loop on the chats to find the right chat for the current message (using the chatId)?

Reply:
I thought about it... and think you are right